src/Functions/getSetting.cpp

```diff
@@ -60,15 +67,30 @@ class FunctionGetSetting : public IFunction, WithContext
                             String{name});
 
         std::string_view setting_name{column->getDataAt(0).toView()};
-        return getContext()->getSettingsRef().get(setting_name);
+        Field setting_value;
+        if constexpr (mode == ErrorHandlingMode::Exception)
+            setting_value = getContext()->getSettingsRef().get(setting_name);
+        else
+        {
+            const auto * default_value_column = arguments[1].column.get();
+            if (!default_value_column || default_value_column->size() != 1)
+            {
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                                "The 2nd argument of function {} should be a constant with the default value of a setting", String{name});
+            }
+            if (!getContext()->getSettingsRef().tryGet(setting_name, setting_value))
+                setting_value = (*default_value_column)[0];
+        }
+        return setting_value;
     }
 };
 
 }
 
 REGISTER_FUNCTION(GetSetting)
 {
-    factory.registerFunction<FunctionGetSetting>();
+    factory.registerFunction<FunctionGetSetting<ErrorHandlingMode::Exception>>();
+    factory.registerFunction<FunctionGetSetting<ErrorHandlingMode::Default>>();
```

Comment:
One more thing, you also need to add description for these functions now. Here's an example: https://github.com/ClickHouse/ClickHouse/blob/86897226592ff49622472011934cc8f44974123d/src/Functions/array/arrayElement.cpp#L2206-L2226